.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: Build and Deploy Documentation using MkDocs
+
+# Controls when the action will run. Triggers the workflow on push events on any branch 
+# but pull request events only for the main branch
+on:
```

Comment:
Any reason not to run the workflow on any PR to any branch?

Reply:
On second thought let's go with:

```
on:
  push:
    branches:
      - main
  pull_request:
  ```

ie run for all PRs and only on pushes to main branch